JSDemos/Demos/Toolbar/Multiline/React/App.js

```diff
@@ -0,0 +1,317 @@
+import CheckBox from 'devextreme-react/check-box';
+import DateBox from 'devextreme-react/date-box';
+import DropDownButton from 'devextreme-react/drop-down-button';
+import SelectBox from 'devextreme-react/select-box';
+import notify from 'devextreme/ui/notify';
+import 'devextreme/ui/select_box';
+import {
+  fontSizes,
+  lineHeights,
+  fonts,
+  fontStyles,
+  textAligns,
+  listTypes,
+} from './data.js';
+
+function getTextAlignItems(isExtended) {
+  return !isExtended
+    ? textAligns.map(({ icon, alignment, hint }) => ({ icon, alignment, hint }))
+    : textAligns;
+}
+
+const toolbarLineModes = ['Multiline mode', 'Singleline mode'];
+
+const attachButtonOptions = {
+  icon: 'attach',
+  text: 'Attach',
+  onClick: () => {
+    notify('Attach button has been clicked!');
+  },
+};
+
+const addButtonOptions = {
+  icon: 'add',
+  text: 'Add',
+  onClick: () => {
+    notify('Add button has been clicked!');
+  },
+};
+
+const removeButtonOptions = {
+  icon: 'trash',
+  text: 'Remove',
+  onClick: () => {
+    notify('Remove button has been clicked!');
+  },
+};
+
+const aboutButtonOptions = {
+  icon: 'help',
+  text: 'About',
+  onClick: () => {
+    notify('About button has been clicked!');
+  },
+};
+
+function App() {
+  const [lineHeight, setLineHeight] = React.useState(lineHeights[1].lineHeight);
+  const [fontSize, setFontSize] = React.useState(fontSizes[2].size);
+  const [textAlign, setTextAlign] = React.useState([textAligns[0].alignment]);
+  const [toolbarLineMode, setToolbarLineMode] = React.useState(toolbarLineModes[0]);
+  const [multiline, setMultiline] = React.useState(true);
+  const [checkBoxValue, setCheckBoxValue] = React.useState(false);
+
+  const onCheckBoxChange = React.useCallback(
+    ({ value }) => {
+      setCheckBoxValue(value);
+      notify('Private option has been changed!');
+    }, [],
+  );
+
+  const onLineHeightChange = React.useCallback(
+    (e) => setLineHeight(e.item.lineHeight), [],
+  );
+
+  const onFontSizeChange = React.useCallback(
+    (e) => setFontSize(e.item.size), [],
+  );
+
+  const onTextAlignChange = React.useCallback(
+    (e) => setTextAlign([e.item.alignment]), [],
+  );
+
+  const onToolbarLineModeChange = React.useCallback(({ value }) => {
+    setMultiline(value === toolbarLineModes[0]);
+    setToolbarLineMode(value);
+  }, []);
+
+  const renderFontSize = React.useCallback(
+    (itemData) => (
+      <div style={{ fontSize: `${itemData.size}px` }}>{itemData.text}</div>
+    ),
+    [],
+  );
+
+  const renderTextAlign = React.useCallback(() => {
+    const isMenu = false;
+
+    return (
+      <ButtonGroup
+        items={getTextAlignItems(isMenu)}
+        keyExpr="alignment"
+        stylingMode="outlined"
+        selectedItemKeys={textAlign}
+        onSelectionChanged={onTextAlignChange}
+      ></ButtonGroup>
+    );
+  }, [textAlign]);
+
+  const renderTextAlignMenu = React.useCallback(() => {
+    const isMenu = true;
+
+    return (
+      <ButtonGroup
```

Comment:
Are you checked that buttongroup in menu looks good? I think we need additional cssClass property to dxItem with corresponding class. It should work after the follow request will be merged -  https://github.com/DevExpress/DevExtreme/pull/22625